storage/client/src/main/java/com/azure/storage/queue/QueueServiceAsyncClient.java

```diff
@@ -0,0 +1,175 @@
+            if (!StorageErrorCode.QUEUE_NOT_FOUND.toString().equals(ex.value().message())
+                && !StorageErrorCode.QUEUE_BEING_DELETED.toString().equals(ex.value().message())) {
+                throw ex;
+            }
+        }
+    }
+
+    /**
+     * Lists the queues in the storage account
+     * @return queues in the storage account
+     */
+    public Flux<QueueItem> listQueuesSegment() {
+        return listQueuesSegment(null, null);
+    }
+
+    /**
+     * Lists the queues in the storage account
+     * @param options Filter for queue selection
+     * @return queues in the storage account that satisfy the filter requirements
+     */
+    public Flux<QueueItem> listQueuesSegment(QueuesSegmentOptions options) {
+        return listQueuesSegment(null, options);
+    }
+
+    /**
+     * Lists the queues in the storage account
+     * @param marker Starting point to list the queues
+     * @param options Filter for queue selection
+     * @return queues in the storage account that satisfy the filter requirements
+     */
+    Flux<QueueItem> listQueuesSegment(String marker, QueuesSegmentOptions options) {
+        String prefix = null;
+        Integer maxResults = null;
+        List<ListQueuesIncludeType> include = null;
+
+        if (options != null) {
+            prefix = options.prefix();
+            maxResults = options.maxResults();
+            if (options.includeMetadata()) {
+                include = Collections.singletonList(ListQueuesIncludeType.fromString(ListQueuesIncludeType.METADATA.toString()));
+            }
+        }
+
+        return client.services().listQueuesSegmentWithRestResponseAsync(prefix, marker, maxResults, include, null, null, Context.NONE)
+            .flatMapMany(response -> Flux.fromIterable(response.value().queueItems()));
+    }
+
+    /**
+     * @return Global queue properties in the storage account
+     */
+    public Mono<Response<StorageServiceProperties>> getProperties() {
+        return client.services().getPropertiesWithRestResponseAsync(Context.NONE)
+            .map(response -> new SimpleResponse<>(response.request(), response.statusCode(), response.headers(), response.value()));
+    }
+
+    /**
+     * Sets global queue properties for the storage account
+     * @param properties Queue properties
+     * @return an empty response
+     */
+    public Mono<VoidResponse> setProperties(StorageServiceProperties properties) {
+        return client.services().setPropertiesWithRestResponseAsync(properties, Context.NONE)
+            .map(VoidResponse::new);
+    }
+
+    /**
+     * @return global statics about queues in the storage account
+     */
+    public Mono<Response<StorageServiceStats>> getStatistics() {
+        return client.services().getStatisticsWithRestResponseAsync(Context.NONE)
+            .map(response -> new SimpleResponse<>(response.request(), response.statusCode(), response.headers(), response.value()));
```

Comment:
I hate this line - I see it everywhere.

Reply:
I'll make a helper method, a lot of the returns need this type of transformation.